tests/material_derivatives_simpler.cc

```diff
@@ -0,0 +1,95 @@
+// make sure we use the additional material model outputs
+#include <aspect/simulator.h>
+#include <deal.II/grid/tria.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/material_model/simple.h>
+#include <aspect/simulator_access.h>
+
+#include <iostream>
+#include <typeinfo>
+
+
+int f()
+{
+  const int dim=2;
+std::cout << "Flag 1" << std::endl;
+  using namespace aspect::MaterialModel;
+  MaterialModelInputs<dim> in_basis(3,3);
+  in_basis.composition[0][0] = 1;
+  in_basis.composition[0][1] = 0;
+  in_basis.composition[0][2] = 0;
+  in_basis.composition[1][0] = 0.5;
+  in_basis.composition[1][1] = 0.25;
+  in_basis.composition[1][2] = 0.25;
+  in_basis.composition[2][0] = 0;
+  in_basis.composition[2][1] = 0.5;
+  in_basis.composition[2][2] = 0.5;
+
+  in_basis.pressure[0] = 10;
+  in_basis.pressure[1] = 1000;
+  in_basis.pressure[2] = 1e6;
+
+  in_basis.strain_rate[0] = SymmetricTensor<2,dim>(); // What do I fill it with?
+  in_basis.strain_rate[1] = SymmetricTensor<2,dim>(); // What do I fill it with?
+  in_basis.strain_rate[2] = SymmetricTensor<2,dim>(); // What do I fill it with?
+
+  in_basis.temperature[0] = 293;
+  in_basis.temperature[1] = 1000;
+  in_basis.temperature[2] = 2000;
+
+
+  MaterialModelInputs<dim> in_dp(3,3);
+  in_dp = in_basis;
+  in_dp.pressure[0] *=1.0000000001;
+  in_dp.pressure[1] *=1.0000000001;
+  in_dp.pressure[2] *=1.0000000001;
+  MaterialModelInputs<dim> in_dstrainrate(3,3);
+  in_dstrainrate = in_basis;
+  in_dstrainrate.strain_rate[0] *=1.0000000001;
+  in_dstrainrate.strain_rate[1] *=1.0000000001;
+  in_dstrainrate.strain_rate[2] *=1.0000000001;
```

Comment:
This multiplies *all* components of the strain rate tensor by the same factor. But you probably want to test derivatives with regard to the components of the strain rate *separately*, so you need `dim*(dim+1)/2` different `in_dstrainrate` objects, and in each of them you multiply only one component by 1.0000001.